src/NLog/Layouts/LayoutParser.cs

```diff
@@ -457,7 +460,10 @@ private static void SetDefaultPropertyValue(ConfigurationItemFactory configurati
             }
             else
             {
-                // TODO NLog 5.0 Should throw exception when invalid configuration (Check throwConfigExceptions)
+                if (throwConfigExceptions ?? LogManager.ThrowConfigExceptions == true)
+                {
+                    throw new NLogConfigurationException("{0} has no default property to assign value {1}", layoutRenderer.GetType(), value);
```

Comment:
Instead of having the same log formatting twice, then one could consider this:

```c#
var configException = new NLogConfigurationException($"{layoutRenderer.GetType()} has no default property to assign value {value}");
if (throwConfigExceptions ?? LogManager.ThrowConfigExceptions ?? LogManager.ThrowExceptions)
     throw configException;

InternalLogger.Warn(configException.Message);
```

Reply:
Here it's not a rethrow unless I'm mistaken about something. Also, if the `InternalLogger.Warn` is removed then nothing will be logged at all if throwing isn't enabled right?

```c#
var configException = new NLogConfigurationException($"{layoutRenderer.GetType()} has no default property to assign value {value}");
if (throwConfigExceptions ?? configException.MustBeRethrown())
     throw configException;
InternalLogger.Warn("{0} has no default property to assign value {1}", layoutRenderer.GetType(), value);
```

Reply:
@njqdev The `MustBeRethrown()`-extension-method will perform logging to the InternalLogger.